lab1/myswitch_stp.py

```diff
@@ -125,6 +131,8 @@ def handle_stm(net, interfaces, stp_context, pkt, incoming_interface):
     stm.hops_to_root += 1
 
     def update_info_as_described_in_point_4_and_forward_STP():
+        if stm.root < stp_context.my_id:
+            stp_context.set_root(stm.root)
         stp_context.root_interface = incoming_interface
         stp_context.root_switch_id = stm.switch_id
         stp_context.hops_from_root = stm.hops_to_root
```

Comment:
Below, we should quit without forwarding before checking _if stm.root == stp_context.root_id_ if the current switch is still root. Otherwise we could end up blocking an interface on root if one if its STM packets get back to it.

Reply:
Hmm, I don't know about this one (or maybe I don't quite understand).

If we don't forward, a downstream non-root switch may be "starved" for the 10 seconds threshold, and suddenly declare itself root.

Furthermore, with my update 663f34a7 just now, root can no longer block any interfaces.

Reply:
Okay, I'll take it